### DISCUSSION TITLE
Multiple loggers using the same file sink

### DISCUSSION THREAD

**Discussion by nimcomputing** [SpriteOvO/spdlog-rs#21] [Q&A]
Hi,

Is it possible to use the same rotating file sink for multiple loggers?

I've an application that's split into multiple components which I'd like a separate logger for and the components are each within a separate module.

I'm thinking the loggers themselves will be stored in a logging module, within mutex locked hashmap.

Thanks in advance.
Luke

**Comment by SpriteOvO** [comment]
> Is it possible to use the same rotating file sink for multiple loggers?

Yes, obviously you can use the same sink for multiple loggers. that's why `Logger` accepts `Arc<Sink>`.

> I've an application that's split into multiple components which I'd like a separate logger for and the components are each within a separate module.

This is an expected usage. The difference between `spdlog::Logger` and the official Rust `log::Logger` is intended for this flexibility.

> I'm thinking the loggers themselves will be stored in a logging module, within mutex locked hashmap.

Currently there is no such thing as a logger registry, so it will not be stored anywhere by `spdlog-rs` other than the object you build from the builder.
